addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -101,50 +131,78 @@ def _get_form_fields_mapping(self, order, doc_line_id_mapping=None):
+            for field in all_form_fields
         }
 
-        # Adding fields from each line, prefixed by the line_id to avoid conflicts
-        lines_with_doc_ids = set(doc_line_id_mapping.values())
-        for line in order.order_line.filtered(lambda sol: sol.id in lines_with_doc_ids):
-            form_fields_mapping.update(self._get_sol_form_fields_mapping(line))
-
         return form_fields_mapping
 
-    def _get_sol_form_fields_mapping(self, line):
-        """ Dictionary mapping specific pdf fields name to Odoo fields data for a sale order line.
-
-        Fields name are prefixed by the line id to avoid conflict between files.
+    def _get_formatted_field(self, form_field_name, order, translated_env, tz, param_field_map):
+        """ Format a field value from the extracted PDF string.
 
-        Override this method to add new fields to the mapping.
+        :param string form_field_name: field path
+        :param recordset order: sale.order record from which to take the value, following the path
+        :param dict translated_env: self.env, with the correct translation context to format amount,
+                                    date and datetime
+        :param str tz: the timezone used for rendering datetime
+        :param dict param_field_map: the map stored in the config parameter to assign the expected
+                                     paths to the form fields. A form field not in that map will be
+                                     ignored.
+        :rtype: string
+        :return: formatted field value
 
-        :param recordset line: sale.order.line record
-        :rtype: dict
-        :return: mapping of prefixed fields name to Odoo fields data
-
-        Note: line.ensure_one()
+        Note: order.ensure_one()
         """
-        line.ensure_one()
-        env = self.with_context(use_babel=True).env
-        return {
-            f'{line.id}_description': line.name,
-            f'{line.id}_quantity': line.product_uom_qty,
-            f'{line.id}_uom': line.product_uom.name,
-            f'{line.id}_price_unit': format_amount(env, line.price_unit, line.currency_id),
-            f'{line.id}_discount': line.discount,
-            f'{line.id}_product_sale_price': format_amount(
-                env, line.product_id.lst_price, line.product_id.currency_id
-            ),
-            f'{line.id}_taxes': ', '.join(tax.name for tax in line.tax_id),
-            f'{line.id}_tax_excl_price': format_amount(env, line.price_subtotal, line.currency_id),
-            f'{line.id}_tax_incl_price': format_amount(env, line.price_total, line.currency_id),
-        }
+        def _get_formatted_value(field_type_, value_):
+            if field_type_ == 'boolean':
+                formatted_value = _("Yes") if value_ else _("No")
+            elif field_type_ == 'monetary':
+                currency_id = records[field.get_currency_field(records)]
```

Comment:
records ?
Seems like a potential issue here, you want to have the currency of the specific record whose field is being formatted, not the batch of records.

Reply:
Hmm, indeed, could I go the simplest way of taking the currency from the same record? :thinking: 
That could be potentially wrong, but I see no case with a good configuration and a dynamic field that makes sense where the currencies could differ... :thinking:

Reply:
Can't you give the field_name and record to the method, instead of the value to be formatted ?

Reply:
=> I simply took it from the `record` instead of the `records` =]